internal/services/mssql/mssql_database_resource.go

```diff
@@ -488,133 +492,175 @@ func resourceMsSqlDatabaseRead(d *pluginsdk.ResourceData, meta interface{}) erro
+				if err := d.Set("long_term_retention_policy", helper.FlattenLongTermRetentionPolicy(model)); err != nil {
+					return fmt.Errorf("setting `long_term_retention_policy`: %+v", err)
+				}
+			}
+
+			shortTermPolicy, err := shortTermRetentionClient.Get(ctx, pointer.From(id))
+
+			if model := shortTermPolicy.Model; model != nil {
+				if err != nil {
+					return fmt.Errorf("retrieving Short Term Retention Policies for %s: %+v", id, err)
+				}
+
+				if err := d.Set("short_term_retention_policy", helper.FlattenShortTermRetentionPolicy(model)); err != nil {
+					return fmt.Errorf("setting `short_term_retention_policy`: %+v", err)
+				}
+			}
+		} else {
+			// DW SKUs need the retention policies to be empty for state consistency
+			emptySlice := make([]interface{}, 0)
+			d.Set("long_term_retention_policy", emptySlice)
+			d.Set("short_term_retention_policy", emptySlice)
+
+			geoPoliciesResponse, err := geoBackupPoliciesClient.Get(ctx, pointer.From(id))
+			if err != nil {
+				if response.WasNotFound(geoPoliciesResponse.HttpResponse) {
+					d.SetId("")
+					return nil
+				}
+
+				return fmt.Errorf("retrieving Geo Backup Policies for %s: %+v", id, err)
+			}
+
+			// For Datawarehouse SKUs, set the geo-backup policy setting
+			if model := geoPoliciesResponse.Model; model != nil {
+				if strings.HasPrefix(skuName, "DW") && model.Properties.State == geobackuppolicies.GeoBackupPolicyStateDisabled {
+					geoBackupPolicy = false
+				}
+			}
 		}
 	}
 
 	if err := d.Set("geo_backup_enabled", geoBackupPolicy); err != nil {
 		return fmt.Errorf("setting `geo_backup_enabled`: %+v", err)
 	}
 
-	tde, err := transparentEncryptionClient.Get(ctx, id.ResourceGroup, id.ServerName, id.Name)
+	securityAlertPolicy, err := securityAlertPoliciesClient.Get(ctx, pointer.From(id))
+	if err == nil && securityAlertPolicy.Model != nil {
+		if err := d.Set("threat_detection_policy", flattenMsSqlServerSecurityAlertPolicy(d, pointer.From(securityAlertPolicy.Model))); err != nil {
+			return fmt.Errorf("setting `threat_detection_policy`: %+v", err)
+		}
+	}
+
+	tde, err := transparentEncryptionClient.Get(ctx, pointer.From(id))
 	if err != nil {
-		return fmt.Errorf("while retrieving Transparent Data Encryption status of %q: %+v", id.String(), err)
+		return fmt.Errorf("while retrieving Transparent Data Encryption state for %s: %+v", id, err)
 	}
-	tdeStatus := false
-	if tde.TransparentDataEncryptionProperties != nil && tde.TransparentDataEncryptionProperties.Status == sql.TransparentDataEncryptionStatusEnabled {
-		tdeStatus = true
+
+	tdeState := false
+	if model := tde.Model; model != nil {
+		if props := model.Properties; props != nil {
+			tdeState = (props.State == transparentdataencryptions.TransparentDataEncryptionStateEnabled)
+		}
 	}
-	d.Set("transparent_data_encryption_enabled", tdeStatus)
 
-	return tags.FlattenAndSet(d, resp.Tags)
+	d.Set("transparent_data_encryption_enabled", tdeState)
+
+	return tags.FlattenAndSet(d, resp.Model.Tags)
```

Comment:
we can move this up:

```suggestion
	return nil
```

Reply:
I am not sure what you mean by this?

Reply:
Fixed.